elyra/pipeline/component_parser_kfp.py

```diff
@@ -140,11 +167,67 @@ def _get_adjusted_parameter_fields(self,
                 for command in component_body['implementation']['container']['command']:
                     if isinstance(command, dict) and list(command.values())[0] == io_object_name and \
                             list(command.keys())[0] == f"{io_object_type}Path":
-                        adjusted_type = "file"
+                        adjusted_type = f"{io_object_type}Path"
             if "args" in component_body['implementation']['container']:
                 for arg in component_body['implementation']['container']['args']:
                     if isinstance(arg, dict) and list(arg.values())[0] == io_object_name and \
                             list(arg.keys())[0] == f"{io_object_type}Path":
-                        adjusted_type = "file"
+                        adjusted_type = f"{io_object_type}Path"
 
         return adjusted_type
+
+    def determine_type_information(self, parsed_type: str) -> Tuple[str, str, str, bool, Any]:
+        """
+        Takes the type information of a component parameter as parsed from the component
+        specification and returns a new type that is one of several standard options.
+
+        """
+        type_lowered = parsed_type.lower()
+        type_options = ['dictionary', 'dict', 'set', 'list', 'array', 'arr', 'inputpath', 'inputvalue', 'outputpath']
+
+        # Prefer types that occur in a clause of the form "[type] of ..."
+        # E.g. "a dictionary of key/value pairs" will produce the type "dictionary"
+        for option in type_options:
+            if any(word + " of " in type_lowered for word in type_options):
+                reg = re.compile(f"({option}) of ")
+                match = reg.search(type_lowered)
+                if match:
+                    type_lowered = option
+                    break
+            elif option in type_lowered:
+                type_lowered = option
+                break
+
+        # Set control id and default value for UI rendering purposes
+        # Standardize type names
+        control = "custom"
+        control_id = "StringControl"
+        default_value = ''
+        required_param = True
+        if any(word in type_lowered for word in ["str", "string"]):
+            type_lowered = "string"
+        elif any(word in type_lowered for word in ['int', 'integer', 'number']):
+            type_lowered = "number"
+            control_id = "NumberControl"
+            default_value = 0
+        elif any(word in type_lowered for word in ['bool', 'boolean']):
+            type_lowered = "boolean"
+            control_id = "BooleanControl"
+            default_value = False
+        elif type_lowered in ['dict', 'dictionary']:
+            type_lowered = "dictionary"
+        elif type_lowered in ['list', 'set', 'array', 'arr']:
+            type_lowered = "list"
+        elif type_lowered in ['inputpath']:
```

Comment:
This can only be `type_lowered == 'inputpath'` and reads better.  Same goes for `inputvalue`, `outputvalue` and `file` in superclass.